xbmc/addons/AddonDatabase.cpp

```diff
@@ -93,9 +96,13 @@ static void DeserializeMetadata(const std::string& document, CAddonBuilder& buil
   builder.SetPackageSize(variant["size"].asUnsignedInteger());
 
   builder.SetPath(variant["path"].asString());
-  builder.SetFanart(variant["fanart"].asString());
   builder.SetIcon(variant["icon"].asString());
 
+  std::map<std::string, std::string> art;
+  for (auto it = variant["art"].begin_map(); it != variant["art"].end_map(); ++it)
```

Comment:
`CVariant` doesn't work like this if it doesn't contain `art`, because you get independent iterators for `begin_map()` and `end_map()`.
It must be enclosed into `if (!variant["art"].empty())`